src/clib/pioc_support.cpp

```diff
@@ -553,46 +553,115 @@ adios2_variable* spio_define_adios2_variable(iosystem_desc_t *ios, file_desc_t *
 #ifdef _SPIO_ADIOS_USE_COMPRESSION
     assert(ios != NULL && file != NULL);
     /* Skip compression for scalar variables (ndims == 0), see https://github.com/ornladios/ADIOS2/issues/4390 */
-    if (variable != NULL && ios->compression_operator != NULL && ndims > 0 && (file->iotype == PIO_IOTYPE_ADIOSC))
+    if (variable != NULL && ndims > 0 && (file->iotype == PIO_IOTYPE_ADIOSC))
     {
         size_t operation_index = 0;
         adios2_error adiosErr = adios2_error_none;
 
-        #if defined(ADIOS2_HAVE_BLOSC2)
-        /* Add a compression operation (Blosc2) to this variable */
-        adiosErr = adios2_add_operation(&operation_index, variable, ios->compression_operator, "compressor", "zstd"); /* Compressor type (e.g., zstd) */
-        if (adiosErr != adios2_error_none)
+        /* Use lossless compression method by default */
+        int variable_compression_method = ios->adios_lossless_compression_method;
+
+        #ifdef _SPIO_ADIOS_USE_LOSSY_COMPRESSION
+        /* Switch to lossy compression method */
+        variable_compression_method = ios->adios_lossy_compression_method;
+
+        /* SZ or MGARD compressor only supports float or double types */
+        if (variable_compression_method == ADIOS_COMPRESSION_METHOD_SZ || variable_compression_method == ADIOS_COMPRESSION_METHOD_MGARD)
         {
-            pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
-                    "Failed to add Blosc2 compression operation (compressor=zstd) to variable %s (adios2_error=%s)",
-                    name, convert_adios2_error_to_string(adiosErr));
+            /* Switch back to lossless compression method for non floating point type */
+            if (type != adios2_type_float && type != adios2_type_double)
```

Comment:
We need to add info on this switch during configure

Reply:
The type of the variable is only known during run time, not configuration time. Also, this switch is only locally for this specific variable (non-floating-point).

Reply:
OK, in the CMake configuration message we can mention this this possible switch.